src/endless/UEFIBootSetup.cpp

```diff
@@ -460,4 +460,49 @@ bool EFICreateNewEntry(const wchar_t *drive, wchar_t *path, wchar_t *desc) {
 
 error: 
 	return retResult;
+}
+
+bool EFIRemoveEntry(wchar_t *desc) {
+	wchar_t varname[9];
+	int target = -1;
+
+	//print_entries();
+
+	target = EFIGetBootEntryNumber(desc);
+	IFFALSE_RETURN_VALUE(target != -1, "Failed to find EFI entry for Endless OS", false);
+
+	swprintf(varname, sizeof(varname), UEFI_VAR_BOOT_ENTRY_FORMAT, target);
+	/* Writing a zero length variable deletes it */
+	IFFALSE_RETURN_VALUE(SetFirmwareEnvironmentVariable(varname, L"{8BE4DF61-93CA-11d2-AA0D-00E098032B8C}", NULL, 0), "Error on SetFirmwareEnvironmentVariable", false);
```

Comment:
Why not use `UEFI_BOOT_NAMESPACE` here and a couple of other places? Is this a wide vs narrow string thing?